src/components/PublishingTopbar/PublishingTopbar.tsx

```diff
@@ -18,53 +20,53 @@ import {
   StyledLink,
 } from './PublishingTopbar.style'
 
-type Channel = {
-  name: string
-  avatar: string
-}
 type ChannelInfoProps = {
   active?: boolean
-  member: string
-  channel: Channel
+  memberName?: string
+  channel: BasicChannelFieldsFragment | undefined
   onClick?: React.MouseEventHandler<HTMLDivElement>
 }
 
 type MemberInfoProps = {
-  memberName: string
-  memberAvatar: string
+  member: BasicMembershipFieldsFragment | undefined
 }
 
 type NavDrawerProps = {
   active?: boolean
-  memberName: string
-  memberAvatar: string
-  channels: Channel[]
-  currentChannel: Channel
-  onCurrentChannelChange: (channel: Channel) => void
+  channels: BasicChannelFieldsFragment[] | undefined
+  currentChannel: BasicChannelFieldsFragment | undefined
+  onCurrentChannelChange: (channelId: string) => void
   onLogoutClick: () => void
   handleClose: () => void
-}
-
-const member = {
-  name: 'Mikael Cowan',
-  avatar: 'https://picsum.photos/300/300',
-  channels: [
-    { name: 'Wild Crypto Fan16', avatar: 'https://picsum.photos/201/300' },
-    { name: 'Mild Crypto Skeptic', avatar: 'https://picsum.photos/202/300' },
-    { name: 'Average Cryptocurrency Enjoyer', avatar: 'https://picsum.photos/203/300' },
-  ],
-}
+} & MemberInfoProps
 
 const PublishingTopbar: React.FC = () => {
   const { activeUser, setActiveChannel } = useActiveUser()
-  // TODO Add member fetching
-  const [currentChannel, setCurrentChannel] = useState(member.channels[0])
-
+  const { memberships, loading, error } = useMemberships(
```

Comment:
I think here we actually can use `useMembership`. We can assume the membership will be set somewhere else (sign in view). So here we should just fetch one specific membership with ID specified in `activeUser`